include/picongpu/plugins/saxs/Saxs.kernel

```diff
@@ -0,0 +1,368 @@
+     */
+    template< uint32_t T_numWorkers >
+    struct KernelSaxs
+    {
+        /**
+         * The SAXS kernel calculates for all particles on the device the
+         * scattering intensity for input calculation ranges.
+         * The parallelization is as follows:
+         *  - The number of threads per block is equal to the number of cells per
+         *    super cells which is also equal to the number of particles per frame
+         *
+         * The procedure starts with calculating unique ids for the threads and
+         * initializing the shared memory.
+         * Then a loop over all super cells starts.
+         * Every thread loads a particle from that super cell and calculates its
+         * scattering structure factor.
+         * For every Particle
+         * exists therefore a unique space within the shared memory.
+         * After that, a thread calculates for a specific scattering vector of all
+         * particles.
+         */
+        template <
+            typename ParBox,
+            typename DBox,
+            typename DBox_np,
+            typename DBox_nmp,
+            typename Mapping,
+            typename T_Acc>
+        DINLINE
+        void
+        operator()(
+            T_Acc const &acc,
+            ParBox pb,
+            DBox sumfcoskr,
+            DBox sumfsinkr,
+            DBox_np np,
+            DBox_nmp nmp,
+            DataSpace< simDim > globalOffset,
+            Mapping mapper,
+            ReciprocalSpace const & reciprocalSpace
+        ) const
+        {
+            constexpr uint32_t frameSize =
+                pmacc::math::CT::volume< SuperCellSize >::type::value; /// fix computation from particle box
+
+            // vectorial part of the integrand in the Jackson formula
+            using PositionArray = memory::Array<
+                float3_X,
+                frameSize
+            >;
+            PMACC_SMEM(
+                acc,
+                cachedR,
+                PositionArray
+            );
+
+            using WeightingArray = memory::Array<
+                float_X,
+                frameSize
+            >;
+            PMACC_SMEM(
+                acc,
+                cachedWeighting,
+                WeightingArray
+            );
+
+
+            /* number of super cells on GPU per dimension (still including guard
+             * cells) remove both guards from count [later one sided guard needs to
+             * be added again]
+             */
+            auto const guardingSuperCells = mapper.getGuardingSuperCells();
+            auto const superCellsCount =
+                mapper.getGridSuperCells() - 2 * guardingSuperCells;
+
+            // get absolute number of relevant super cells
+            auto const numSuperCells = superCellsCount.productOfComponents();
+
+            detail::Result result;
+            for( auto supercellLinearIdx = 0; supercellLinearIdx < numSuperCells; ++supercellLinearIdx )
```

Comment:
please use a explicit type to better understand which type is used for the index, currently it is `int` because of the signed zero.